src/coreclr/jit/lclvars.cpp

```diff
@@ -2914,15 +2914,17 @@ void Compiler::lvaSetStruct(unsigned varNum, ClassLayout* layout, bool unsafeVal
 
         if (layout->IsValueClass())
         {
-            varDsc->lvType = layout->GetType();
+            varDsc->lvType                          = layout->GetType();
+            CORINFO_CLASS_HANDLE const classHandle  = layout->GetClassHandle();
+            unsigned const             corinfoFlags = this->info.compCompHnd->getClassAttribs(classHandle);
+            varDsc->SetIsSpan(corinfoFlags & CORINFO_FLG_SPAN);
```

Comment:
This should be unified with the code below that calls `getClassAttribs`.

Note that the code below was refactored a bit to only call `getClassAttribs` when `unsafeValueClsCheck` is true; it's fine to turn it back to look more like what it was before.

Reply:
Missed the fact the same call was made below. Sorry.